lib/src/database/sqflite_box.dart

```diff
@@ -0,0 +1,274 @@
+  bool get _keysCached => _cachedKeys != null;
+
+  static const Set<Type> allowedValueTypes = {
+    List<dynamic>,
+    Map<dynamic, dynamic>,
+    String,
+    int,
+    double,
+    bool,
+  };
+
+  Box(this.name, this.boxCollection) {
+    if (!allowedValueTypes.any((type) => V == type)) {
+      throw Exception(
+        'Illegal value type for Box: "${V.toString()}". Must be one of $allowedValueTypes',
+      );
+    }
+  }
+
+  String? _toString(V? value) {
+    if (value == null) return null;
+    switch (V) {
+      case const (List<dynamic>):
+      case const (Map<dynamic, dynamic>):
+        return jsonEncode(value);
+      case const (String):
+      case const (int):
+      case const (double):
+      case const (bool):
+      default:
+        return value.toString();
+    }
+  }
+
+  V? _fromString(Object? value) {
+    if (value == null) return null;
+    if (value is! String) {
+      throw Exception(
+          'Wrong database type! Expected String but got one of type ${value.runtimeType}');
+    }
+    switch (V) {
+      case const (int):
+        return int.parse(value) as V;
+      case const (double):
+        return double.parse(value) as V;
+      case const (bool):
+        return (value == 'true') as V;
+      case const (List<dynamic>):
+      case const (Map<dynamic, dynamic>):
+        return jsonDecode(value) as V;
+      case const (String):
+      default:
+        return value as V;
+    }
+  }
+
+  Future<List<String>> getAllKeys([Transaction? txn]) async {
+    if (_keysCached) return _cachedKeys!.toList();
+
+    final executor = txn ?? boxCollection._db;
+
+    final result = await executor.query(name, columns: ['k']);
+    final keys = result.map((row) => row['k'] as String).toList();
+
+    _cachedKeys = keys.toSet();
+    return keys;
+  }
+
+  Future<Map<String, V>> getAllValues([Transaction? txn]) async {
+    final executor = txn ?? boxCollection._db;
+
+    final result = await executor.query(name);
+    return Map.fromEntries(
+      result.map(
+        (row) => MapEntry(
+          row['k'] as String,
+          _fromString(row['v']) as V,
+        ),
+      ),
+    );
+  }
+
+  Future<V?> get(String key, [Transaction? txn]) async {
+    if (_cache.containsKey(key)) return _cache[key];
+
+    final executor = txn ?? boxCollection._db;
+
+    final result = await executor.query(
+      name,
+      columns: ['v'],
+      where: 'k = ?',
+      whereArgs: [key],
+    );
+
+    final value = result.isEmpty ? null : _fromString(result.single['v']);
+    _cache[key] = value;
+    return value;
+  }
+
+  Future<List<V?>> getAll(List<String> keys, [Transaction? txn]) async {
+    if (!keys.any((key) => !_cache.containsKey(key))) {
+      return keys.map((key) => _cache[key]).toList();
+    }
+
+    final executor = txn ?? boxCollection._db;
+
+    final list = <V?>[];
+    final result = await executor.query(
+      name,
+      where: 'k IN (${keys.map((_) => '?').join(',')})',
```

Comment:
Afaik this fails, when you have more than 1000 keys?

Reply:
Should not happen in our case but I added a recursive algorithm to handle this with a comment above

Reply:
Seems like the actual limit is a bit different from what I guessed, but your implementation is still below that:

> To prevent excessive memory allocations, the maximum value of a host parameter number is SQLITE_MAX_VARIABLE_NUMBER, which defaults to 999 for SQLite versions prior to 3.32.0 (2020-05-22) or 32766 for SQLite versions after 3.32.0. 

https://www.sqlite.org/limits.html

Thanks :)